cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -605,6 +605,33 @@ a|
 Not a syntax change but a semantic one.
 The `EXECUTE BOOSTED` privilege will no longer include an implicit `EXECUTE` privilege when granted.
 That means that to execute a procedure or a function with boosted privileges both `EXECUTE` and `EXECUTE BOOSTED` are needed.
+
+a|
+label:functionality[]
+label:updated[]
+[source, cypher, role="noheader"]
+----
+MATCH (n)
+RETURN
+CASE n.prop
+    WHEN null THEN 'one'
+    ELSE 'two'
+END
+----
+Previously, if n.prop is null, 'one' would be returned. Now 'two' is returned.
+a|
+This is a semantic change only.
+Since null = null returns false in Cypher, a when expression no longer matches on null.
+If matching on null is desired, please use `IS NULL` instead.
+[source, cypher, role="noheader"]
+----
+MATCH (n)
+RETURN
+CASE
+    WHEN n.prop IS NULL THEN 'one'
+    ELSE 'two'
+END
+----
```

Comment:
```suggestion
a|
label:functionality[]
label:updated[]
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE n.prop
    WHEN null THEN 'one'
    ELSE 'two'
END
----
a|
Previously, if `n.prop` is `null`, `'one'` would be returned. 
Now, `'two'` is returned.

This is a semantic change only.
Since `null = null` returns `false` in Cypher, a `WHEN` expression no longer matches on `null`.

If matching on `null` is required, please use `IS NULL` instead:
[source, cypher, role="noheader"]
----
MATCH (n)
RETURN
CASE
    WHEN n.prop IS NULL THEN 'one'
    ELSE 'two'
END
----
```